src/main/java/miku/command/MarkCommand.java

```diff
@@ -0,0 +1,39 @@
+package miku.command;
+
+import miku.utility.Storage;
+import miku.utility.TaskList;
+import miku.utility.UI;
+
+
+
+/**
+ * Marks a specified task as completed
+ */
+public class MarkCommand extends Command {
+    private int index;
+
+    /**
+     * Initialises a mark command.
+     *
+     * @param index the index of the task to be marked
+     */
+    public MarkCommand(int index) {
+        this.index = index;
+    }
+
+
+    /**
+     * Executes the command to add a task to a task list
+     *
+     * @param taskList The task list that takes in the new task
+     * @param ui       The UI to perform printing
+     * @param storage  The storage function to store the data into a text file
+     */
+    @Override
+    public void execute(TaskList taskList, UI ui, Storage storage) {
+        taskList.mark(index);
+        storage.saveFromTaskList(taskList);
+        ui.printSectionBreak();
+    }
```

Comment:
You've added the try, catch blocks for invalid indexes in other Command classes, you may want to consider adding it here too.